### DISCUSSION TITLE
ValueError: 'S' is not in list

### DISCUSSION THREAD

**Discussion by phyoung123** [deepmodeling/dpgen#673] [Q&A] [answered]
Dear all,
I have successfully tested the CH4 example provided in the dpGEN's Github, but an error occurs when i turn the focus to Si and Al, the error infos are 'ValueError: 'S' is not in list' for Si, and 'ValueError: 'A' is not in list' for Al. I cannot figure out what parameter is improper in my input files, any advices provided here are appreciated.

here is the param.json

{
    "type_map": [
        "Si"
    ],
    "mass_map": [
        28
    ],
    "init_data_prefix": "./",
    "init_data_sys": [
        "POSCAR.02x02x02/02.md/sys-0016/deepmd"
    ],
    "sys_configs_prefix": "./",
    "sys_configs": [
        [
            "POSCAR.02x02x02/01.scale_pert/sys-0016/scale-1.000/00000*/POSCAR"
        ],
        [
            "POSCAR.02x02x02/01.scale_pert/sys-0016/scale-1.000/00001*/POSCAR"
        ]
    ],
    "_comment": " that's all ",
    "numb_models": 4,
    "default_training_param": {
        "model": {
            "type_map": [
                "Si"
            ],
            "descriptor": {
                "type": "se_a",
                "sel": [
                    30
                ],
                "rcut_smth": 0.5,
                "rcut": 5.0,
                "neuron": [
                    25,
                    50,
                    100
                ],
                "resnet_dt": false,
                "axis_neuron": 12,
                "seed": 1
            },
            "fitting_net": {
                "neuron": [
                    120,
                    120,
                    120
                ],
                "resnet_dt": true,
                "seed": 1
            }
        },
        "learning_rate": {
            "type": "exp",
            "start_lr": 0.001,
            "decay_steps": 100,
            "decay_rate":0.95
        },
        "loss": {
            "start_pref_e": 0.02,
            "limit_pref_e": 1,
            "start_pref_f": 1000,
            "limit_pref_f": 1,
            "start_pref_v": 0.0,
            "limit_pref_v": 0.0
        },
        "training": {
            "set_prefix": "set",
            "stop_batch": 2000,
            "batch_size": 1,
            "disp_file": "lcurve.out",
            "disp_freq": 100,
            "numb_test": 4,
            "save_freq": 1000,
            "save_ckpt": "model.ckpt",
            "disp_training": true,
            "time_training": true,
            "profiling": false,
            "profiling_file": "timeline.json",
            "_comment": "that's all"
        }
    },
    "model_devi_dt": 0.002,
    "model_devi_skip": 0,
    "model_devi_f_trust_lo": 0.05,
    "model_devi_f_trust_hi": 0.15,
    "model_devi_e_trust_lo": 10000000000.0,
    "model_devi_e_trust_hi": 10000000000.0,
    "model_devi_clean_traj": true,
    "model_devi_jobs": [
        {
            "sys_idx": [
                0
            ],
            "temps": [
                100
            ],
            "press": [
                1.0
            ],
            "trj_freq": 10,
            "nsteps": 300,
            "ensemble": "nvt",
            "_idx": "00"
        },
        {
            "sys_idx": [
                1
            ],
            "temps": [
                100
            ],
            "press": [
                1.0
            ],
            "trj_freq": 10,
            "nsteps": 3000,
            "ensemble": "nvt",
            "_idx": "01"
        }
    ],
    "fp_style": "vasp",
    "shuffle_poscar": false,
    "fp_task_max": 20,
    "fp_task_min": 5,
    "fp_pp_path": "./",
    "fp_pp_files": [
        "POTCAR"
    ],
    "fp_incar": "./INCAR_fp"
}

the train.log as follow

WARNING:tensorflow:From /apps/users/caep-lzg/package/anaconda/envs/deepmd/lib/python3.9/site-packages/tensorflow/python/compat/v2_compat.py:96: disable_resource_variables (from tensorflow.python.ops.variable_scope) is deprecated and will be removed in a future version.
Instructions for updating:
non-resource variables are not supported in the long term
WARNING:root:Environment variable KMP_BLOCKTIME is empty. Use the default value 0
WARNING:root:Environment variable KMP_AFFINITY is empty. Use the default value granularity=fine,verbose,compact,1,0
/apps/users/caep-lzg/package/anaconda/envs/deepmd/lib/python3.9/importlib/__init__.py:169: UserWarning: The NumPy module was reloaded (imported a second time). This can in some cases result in small but subtle issues and is discouraged.
  _bootstrap._exec(spec, module)
/apps/users/caep-lzg/package/anaconda/envs/deepmd/lib/python3.9/site-packages/deepmd/utils/compat.py:316: UserWarning: It seems that you are using a deepmd-kit input of version 1.x.x, which is deprecated. we have converted the input to >2.0.0 compatible, and output it to file input_v2_compat.json
  warnings.warn(msg)
Traceback (most recent call last):
  File "/apps/users/caep-lzg/package/anaconda/envs/deepmd/bin/dp", line 10, in <module>
    sys.exit(main())
  File "/apps/users/caep-lzg/package/anaconda/envs/deepmd/lib/python3.9/site-packages/deepmd/entrypoints/main.py", line 437, in main
    train_dp(**dict_args)
  File "/apps/users/caep-lzg/package/anaconda/envs/deepmd/lib/python3.9/site-packages/deepmd/entrypoints/train.py", line 91, in train
    jdata = update_sel(jdata)
  File "/apps/users/caep-lzg/package/anaconda/envs/deepmd/lib/python3.9/site-packages/deepmd/entrypoints/train.py", line 341, in update_sel
    descrpt_data = update_one_sel(jdata, descrpt_data)
  File "/apps/users/caep-lzg/package/anaconda/envs/deepmd/lib/python3.9/site-packages/deepmd/entrypoints/train.py", line 317, in update_one_sel
    tmp_sel = get_sel(jdata, rcut)
  File "/apps/users/caep-lzg/package/anaconda/envs/deepmd/lib/python3.9/site-packages/deepmd/entrypoints/train.py", line 280, in get_sel
    _, max_nbor_size = get_nbor_stat(jdata, rcut)
  File "/apps/users/caep-lzg/package/anaconda/envs/deepmd/lib/python3.9/site-packages/deepmd/entrypoints/train.py", line 255, in get_nbor_stat
    train_data = get_data(jdata["training"]["training_data"], max_rcut, type_map, None)
  File "/apps/users/caep-lzg/package/anaconda/envs/deepmd/lib/python3.9/site-packages/deepmd/entrypoints/train.py", line 199, in get_data
    data = DeepmdDataSystem(
  File "/apps/users/caep-lzg/package/anaconda/envs/deepmd/lib/python3.9/site-packages/deepmd/utils/data_system.py", line 79, in __init__
    DeepmdData(
  File "/apps/users/caep-lzg/package/anaconda/envs/deepmd/lib/python3.9/site-packages/deepmd/utils/data.py", line 62, in __init__
    atom_type_ = [type_map.index(self.type_map[ii]) for ii in self.atom_type]
  File "/apps/users/caep-lzg/package/anaconda/envs/deepmd/lib/python3.9/site-packages/deepmd/utils/data.py", line 62, in <listcomp>
    atom_type_ = [type_map.index(self.type_map[ii]) for ii in self.atom_type]
ValueError: 'S' is not in list

Thanks in advance.

caep-feiyang

**Comment by njzjz** [comment, marked as answer]
This should have been fixed in https://github.com/deepmodeling/deepmd-kit/pull/1196.

**Reply by phyoung123** [reply]
Thanks, now it works
